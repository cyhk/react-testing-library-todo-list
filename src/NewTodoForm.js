import React, { Component }from 'react';

class NewTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      task: ""
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    })
  }

  handleSubmit(evt) {
    evt.preventDefault();
    const { task } = this.state
    this.props.addToList({ task });
    this.setState({
      task: ""
    })
  }

  render() {
    const styles = {
      margin: "3px"
    }
    
    return (
    <form>
      <label style={styles} htmlFor="task">Task</label>
      <input style={styles} 
             name="task" 
             id="task"
             value={this.state.task} 
             onChange={this.handleChange} 
             placeholder="What do you need to do?"/>
      <button onClick={this.handleSubmit}>Add</button>
    </form>
    );
  }
}

export default NewTodoForm;
