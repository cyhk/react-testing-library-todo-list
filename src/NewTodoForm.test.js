import React from 'react';
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import NewTodoForm from './NewTodoForm';

it('renders without crashing using shallow', () => {
  shallow(<NewTodoForm />);
});

it('renders without crashing using mount', () => {
  mount(<NewTodoForm />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<NewTodoForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("should change state when input changes", function() {
  let wrapper = mount(<NewTodoForm />);

  //test task
  wrapper
    .find('input[name="task"]')
    .simulate("change", {target: {name: "task", value: "Sleep"}});

  expect(wrapper.state().task).toBe("Sleep");
})

it("should submit when button is pressed", function(){
  const addTodo = jest.fn();
  let wrapper = mount(<NewTodoForm addToList={addTodo}/>);

  wrapper
    .find("button")
    .simulate("click")

  expect(addTodo).toHaveBeenCalled();
});