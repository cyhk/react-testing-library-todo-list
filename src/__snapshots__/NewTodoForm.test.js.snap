// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form>
  <label
    htmlFor="task"
    style={
      Object {
        "margin": "3px",
      }
    }
  >
    Task
  </label>
  <input
    id="task"
    name="task"
    onChange={[Function]}
    placeholder="What do you need to do?"
    style={
      Object {
        "margin": "3px",
      }
    }
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;
